### ISSUE TITLE
replace: invalid expression sre_constants.error: multiple repeat

### ISSUE THREAD

**Issue by mattarchie** [ansible/ansible#65932]
##### SUMMARY
The replace module does not support values for `after`, `before`, and `regexp` that need to be escaped and passed to re.compile as raw strings. I am attempting to automatically maintain my known_hosts file by making sure only the newest ssh host key is present, but since the key hashes can contain sequences like "++", I am unable to properly do so using replace.

It appears the fix would be similar to the fix for #19714 and would be needed on lines 252, 254, and 256 of replace.py. May also want line 250 to use an empty raw string for consistency.

Note: I am unable to currently test on the latest stable and devel, but I didn't compare replace.py on those branches against the stable-2.8 branch and I saw no difference.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- replace

##### ANSIBLE VERSION
```paste below
ansible 2.8.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/matt/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 14:33:56) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### CONFIGURATION
No output from `ansible-config dump --only-changed`.

##### OS / ENVIRONMENT
* Ansible host: Red Hat Enterprise Linux server release 7.7 (Maipo)
* Remote target hosts: Red Hat Enterprise Linux server release 7.6 (Maipo)

##### STEPS TO REPRODUCE
This originally happened when I was trying to provide a host key entry in known_hosts as my value for `before` (not everything time, since not every key is guaranteed to have this issue). However, I confirmed it happens in the much simpler cases listed below, passing "++" to `after`, `before`, or `regexp`.

For the examples below, you must have a text file named "my_file" on the remote host in the home directory of the user ansible is remoting in as. This is because the file's existence is checked before parsing and compiling the regular expressions.

Case 1:
```yaml
- replace:
    path: ~/my_file
    before: "++"
    regexp: "\\S+"
    replace: ""
```

Case 2:
```yaml
- replace:
    path: ~/my_file
    after: "++"
    regexp: "\\S+"
    replace: ""
```

Case 3:
```yaml
- replace:
    path: ~/my_file
    regexp: "++"
    replace: ""
```

##### EXPECTED RESULTS
Assume `my_file` contains the following:
```
this
is
++
some
text
```

Case 1 results:
```


++
some
text
```

Case 2 results:
```
this
is
++


```

Case 3 results:
```
this
is

some
text
```

##### ACTUAL RESULTS
The file is never edited due to the `sre_constants.error: multiple repeat` error.

```
fatal: [host1]: FAILED! => changed=false
  module_stderr: |-
    Shared connection to 1.2.3.4 closed.
  module_stdout: |-
    Traceback (most recent call last):
      … (shortened for ease of copying from ansible host)
        imp.load_module('__main__', mod, module, MOD_DESC)
      File "/tmp/ansible_replace_payload_7bd2Vn/__main__.py", line 302, in <module>
      File "/tmp/ansible_replace_payload_7bd2Vn/__main__.py", line 259, in main
      File "/usr/lib64/pythong2.7/re.py", line 190, in compile
        return _compile(pattern, flags)
      File "/usr/lib64/pythong2.7/re.py", line 242, in _compile
        raise error, v # invalid expression
    sre_constants.error: multiple repeat
  msg: |-
    MODULE FAILURE
    see stdout/stderr for the exact error
  rc: 1
```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/files/replace.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/replace.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @EvanK
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
The `+` character has specific meaning in regular expressions, meaning to match 1 or more of the previous character.

As such, if you want to use the literal `+` in a regular expression it must be escaped.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by mattarchie** [comment]
@sivel I hadn't found a way to do the escape automatically, which is why I created the issue. I searched more carefully this time and have found the regex_escape filter. It would be useful to include a reference to this filter in the documentation of the replace module (and others that accept regex arguments) because it is required in some cases (like passing some ssh keys as part of a regex).

Thank you!
